### ISSUE TITLE
Cannot use sudo command inside Nvidia-Docker image

### ISSUE THREAD

**Issue by schroederdewitt** [moby/moby#36730]
Dear All,

I build the following Dockerfile

````
FROM nvidia/cuda:9.0-cudnn7-devel-ubuntu16.04

RUN apt-get update && \
      apt-get -y install sudo

RUN useradd -m docker && echo "docker:docker" | chpasswd && adduser docker sudo

USER docker

CMD /bin/bash
````

using 

> nvidia-docker build -t sudoprob

and when I run it using

> nvidia-docker run -it sudoprob

and, as user docker inside the container, I call 

> sudo apt-get update

then I get the following error:

> sudo: effective uid is not 0, is /usr/bin/sudo on a file system with the 'nosuid' option set or an NFS file system without root privileges?

I did a little bit of research into the issue and had a look at /proc/mount etc, but I am not able to determine whether the root partition / is mounted with nosuid option or not. In any case, I think this is probably something that needs to be fixed by the image developers. 

I need to be able to use sudo because I am running a lot of 3rdparty software install scripts that don't accept to be run as root and any workarounds I can think of require hours of chmodding and chowning of installation files. I don't want to / cannot hack those installation scripts to be ok with using root.

Is there a workaround, or will this be fixed at some point?
I have already asked this question at  [(nvidia-docker github issues)](https://github.com/NVIDIA/nvidia-docker/issues/673) and was referred here.

Thanks a lot and regards

Christian Schroeder de Witt

PS: I am using 

> Docker version 17.12.1-ce, build 7390fc6

**Comment by asbe** [comment]
Christian, this may be related: I happened upon this issue with the same problem. Containers on one host could sudo just fine, but on a nearly equivalent host the error occured. Finally figured out that the cause was because the problematic host used "overlay" as storage driver and the other "aufs". Both docker installations had moved /var/lib/docker to a drive mounted with "nosuid". Turns out "overlay" respects the "nosuid". Just making sure that /var/lib/docker was mounted on a disk with "nosuid" removed resolved all issues. (Curiously - this does not happen with the setup using "aufs")
